include/pybind11/pytypes.h

```diff
@@ -1213,9 +1214,21 @@ class capsule : public object {
 class tuple : public object {
 public:
     PYBIND11_OBJECT_CVT(tuple, object, PyTuple_Check, PySequence_Tuple)
+    /** \rst
+        Creates an empty ``tuple`` with given ``size`` to be later filled in.
+     \endrst */
     explicit tuple(size_t size = 0) : object(PyTuple_New((ssize_t) size), stolen_t{}) {
         if (!m_ptr) pybind11_fail("Could not allocate tuple object!");
     }
+    /** \rst
+        Creates a ``tuple`` from an initializer list of object instances.
```

Comment:
Maybe 
```suggestion
        Creates a ``tuple`` from an initializer list of ``py::object`` instances.
```
just to be explicit? In addition, the added documentation on `make_tuple` in #2840 could be augmented with some remarks on the new ways to initialize a tuple. I didn't bother to document `list` and `set` there, because they are mutable (pybind11 offers ways to modify tuples, but that's besides the point).

Reply:
Sure. Explicit is better than implicit.